src/lib/consensus/proof_of_stake.ml

```diff
@@ -1030,35 +1045,105 @@ module Make (Inputs : Inputs_intf) :
       ( transition |> Snark_transition.blockchain_state
       |> Blockchain_state.ledger_hash )
 
-  let select this that ~logger ~time_received =
+  let select old_state new_state ~logger ~time_received =
```

Comment:
maybe we can do existing / candidate? Makes it more clear when we keep the existing

Reply:
while you're here, can you make these named args `~existing ~candidate`